packages/compiler/lib/src/compilation/ast_hir_lowering/body.dart

```diff
@@ -392,7 +400,8 @@ abstract class Context {
   ) {
     final result = lower(expression);
     if (result is Error) return Error(result.error);
-    final lowered = result.value.toList();
+    final lowered =
+        result.value.distinctBy((it) => it.id.declarationId).toList();
```

Comment:
This basically only takes the first element, as `it.id.declarationId` is the declaration ID of the current body. I don't think this is what we want.